frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { AddButton, Divider, Title } from './style'
+import { PipelineMetricSelection } from './PipelineMetricSelection'
+import { useAppDispatch, useAppSelector } from '@src/hooks'
+import {
+  addADeploymentFrequencySetting,
+  selectDeploymentFrequencySettings,
+} from '@src/context/pipelineMetricsSettings/pipelineMetricsSettingsSlice'
+import { v4 as uuidV4 } from 'uuid'
+
+export const DeploymentFrequencySettings = () => {
+  const dispatch = useAppDispatch()
+  const deploymentFrequencySettings = useAppSelector(selectDeploymentFrequencySettings)
+
+  const handleClick = () => {
+    dispatch(addADeploymentFrequencySetting())
+  }
+
+  return (
+    <>
+      <Divider>
+        <Title>Deployment frequency settings</Title>
+      </Divider>
+      {deploymentFrequencySettings.map((deploymentFrequencySetting, index) => (
+        <PipelineMetricSelection
+          key={uuidV4()}
```

Comment:
UUID as a key of the component is not recommended, coc, it take memory & time to generate it. 

Just use index, and keep [Keys Must Only Be Unique Among Siblings](https://rescript-lang.org/docs/react/latest/arrays-and-keys#keys-must-only-be-unique-among-siblings),